### PR TITLE
Fully reverse swkbd configuration structure and follow-up to swkbd implementation

### PR DESCRIPTION
**Pull request by jduncanator**
This fully parses the software keyboard configuration struct and implements a few extra features into our "simple" keyboard implementation.

Hopefully this enables someone in the future to build proper user-controlled entry for swkbd.

### CODE REVIEW

File: Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/InitialCursorPosition.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Ryujinx.HLE.HOS.Applets.SoftwareKeyboard
+{
+    /// <summary>
+    /// Identifies the initial position of the cursor displayed in the area.
+    /// </summary>
+    internal enum InitialCursorPosition : uint
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/InitialCursorPosition.cs]: `internal` is not necessary (same for the other enums).

File: Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/SoftwareKeyboardApplet.cs [hunk 2]
```diff
@@ -22,7 +22,8 @@ internal class SoftwareKeyboardApplet : IApplet
 
         private SoftwareKeyboardConfig _keyboardConfig;
 
-        private string _textValue = DEFAULT_TEXT;
+        private string  _textValue = DEFAULT_TEXT;
```
Comment (gdkchan) [inline Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/SoftwareKeyboardApplet.cs]:
```suggestion
        private string   _textValue = DEFAULT_TEXT;
```

Comment (gdkchan) [reply Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/SoftwareKeyboardApplet.cs]: Also the constant names should be PascalCase I believe.

File: Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/InitialCursorPosition.cs (hunk 1 above)
Comment (jduncanator) [reply Ryujinx.HLE/HOS/Applets/SoftwareKeyboard/InitialCursorPosition.cs]: Visual Studio does this by default when extracting an enum to a file :shrug:

**Pull request merged**